release/docker/build-image-multi-arch.sh

```diff
@@ -0,0 +1,152 @@
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-t TARBALL\tSpecify multiple opensearch or opensearch-dashboards tarballs, use the same order as the input for '-a' param, e.g. 'opensearch-1.0.0-linux-x64.tar.gz,opensearch-1.0.0-linux-arm64.tar.gz'. You still need to specify the version - this tool does not attempt to parse the filename."
+    echo -e "-h\t\tPrint this message."
+    echo "--------------------------------------------------------------------------"
+}
+
+function cleanup_docker_buildx() {
+    # Cleanup docker buildx
+    echo -e "\n* Cleanup docker buildx"
+    docker buildx use default
+    docker buildx rm $BUILDER_NAME > /dev/null 2>&1
+}
+
+function cleanup_all() {
+    cleanup_docker_buildx
+    File_Delete $DIR
+}
+while getopts ":ht:v:f:p:a:r:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        t)
+            TARBALL=`realpath $OPTARG`
+            ;;
+        v)
+            VERSION=$OPTARG
+            ;;
+        f)
+            DOCKERFILE=$OPTARG
+            ;;
+        p)
+            PRODUCT=$OPTARG
+            ;;
+        a)
+            ARCHITECTURE=$OPTARG
+            ;;
+        r)
+            REPOSITORY=$OPTARG
+            ;;
+        :)
+            echo "-${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate the required parameters to present
+if [ -z "$VERSION" ] || [ -z "$DOCKERFILE" ] || [ -z "$PRODUCT" ] || [ -z "$ARCHITECTURE" ] || [ -z "$REPOSITORY" ]; then
+  echo "You must specify '-v VERSION', '-f DOCKERFILE', '-p PRODUCT', '-a ARCHITECTURE', '-r REPOSITORY'"
+  usage
+  exit 1
+else
+  echo $VERSION $DOCKERFILE $PRODUCT $ARCHITECTURE
+  IFS=', ' read -r -a ARCHITECTURE_ARRAY <<< "$ARCHITECTURE"
+  IFS=', ' read -r -a TARBALL_ARRAY <<< "$TARBALL"
+fi
+
+if [ "$PRODUCT" != "opensearch" ] || [ "$PRODUCT" != "opensearch-dashboards" ]
+then
+    echo "Enter either 'opensearch' or 'opensearch-dashboards' as product name for -p parameter"
+    exit 1
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] || [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "We only support 'x64' and 'arm64' as architecture name for -a parameter"
+    exit 1
+fi
+
+# Warning docker desktop
+if (! docker buildx version)
+then
+    echo -e "\n* You MUST have Docker Desktop to use buildx for multi-arch images."
+    exit 1
+fi
+
+# Prepare docker buildx
+trap cleanup_all TERM INT EXIT
+DIR=`Temp_Folder_Create`
+echo New workspace $DIR
+echo -e "\n* Prepare docker buildx"
+docker buildx use default
+docker buildx create --name $BUILDER_NAME --use
+docker buildx inspect --bootstrap
+
+# Check buildx status
+echo -e "\n* Check buildx status"
+docker buildx ls | grep $BUILDER_NAME
+docker ps | grep $BUILDER_NAME
```

Comment:
What is the expected output from these two commands? If possible, can this script fail if the output is not what we expect?

Reply:
` docker buildx ls | grep $BUILDER_NAME ` will show whether a builder get created with the given name.
` docker ps | grep $BUILDER_NAME ` will show whether a builder container is provisioned for building multi-arch.

Both would fail if there is nothing in the grep:
```
$ echo 123 | grep 111

$ echo $?
1
```

Reply:
OK, got it.